File: Examples/iOS/SongLibraryPlayer/SongLibraryPlayer/ArtistsViewController.m
//
//  ArtistsViewController.m
//  SongLibraryPlayer
//
//  Created by Aurelius Prochazka on 12/26/13.
//  Copyright (c) 2013 Hear For Yourself. All rights reserved.
//

#import "ArtistsViewController.h"
#import "AlbumsViewController.h"
#import <AVFoundation/AVFoundation.h>
#import <MediaPlayer/MediaPlayer.h>

@interface ArtistsViewController ()
@property (strong, nonatomic) NSMutableArray *artistList;
@end

@implementation ArtistsViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;
    self.artistList = [NSMutableArray arrayWithArray:[[MPMediaQuery artistsQuery] collections]];
    [self.tableView reloadData];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    // Return the number of sections.
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView
 numberOfRowsInSection:(NSInteger)section
{
    // Return the number of rows in the section.
    return self.artistList.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView
         cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *CellIdentifier = @"MusicCell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier
                                                            forIndexPath:indexPath];

    NSString *artistName = [[[self.artistList objectAtIndex:indexPath.row] representativeItem] valueForProperty:MPMediaItemPropertyArtist];
    cell.textLabel.text = artistName;
    return cell;
}


#pragma mark - Navigation

// In a story board-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    UITableViewCell *senderCell = (UITableViewCell *)sender;
    AlbumsViewController *vc = [segue destinationViewController];
    vc.artistName = senderCell.textLabel.text;
    vc.title = senderCell.textLabel.text;
}



@end

File: Examples/iOS/SongProcessor/SongProcessor/View Controllers/iTunes Library Access/ArtistsViewController.swift
//
//  ArtistsViewController.swift
//  SongProcessor
//
//  Created by Aurelius Prochazka on 6/22/16.
//  Copyright © 2016 AudioKit. All rights reserved.
//

import UIKit
import AVFoundation
import MediaPlayer

class ArtistsViewController: UITableViewController {
    
    var artistList: [MPMediaItemCollection] = []
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if let list = MPMediaQuery.artists().collections {
            artistList = list
            tableView.reloadData()
        }
        
    }
    
    // MARK: - Table view data source
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView,
                            numberOfRowsInSection section: Int) -> Int {
        return artistList.count
    }
    
    override func tableView(_ tableView: UITableView,
                            cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cellIdentifier = "MusicCell"
        let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier) ?? UITableViewCell(style: .default, reuseIdentifier: cellIdentifier)
        
        // Configure the cell...
        let repItem = artistList[(indexPath as NSIndexPath).row].representativeItem!
        let artistName = repItem.value(forProperty: MPMediaItemPropertyArtist) as! String
        cell.textLabel?.text = artistName
        
        return cell
    }
    
    // MARK: - Navigation
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        
        if segue.identifier == "AlbumsSegue" {
            let senderCell = sender as! UITableViewCell
            let albumsVC = segue.destination as! AlbumsViewController
            albumsVC.artistName = senderCell.textLabel?.text
            albumsVC.title = senderCell.textLabel?.text
        }
    }
    
    
}
